File: question3.cpp
#include<iostream>
using namespace std;
int main(){
    int number;
    cout<<"please enter number of towers: ";
    cin>>number;
    int towers[number];
    int signalPowers[number];
    for(int i = 0;i<number;++i){
        int input;
        cout<<"enter tower number "<<i+1<<": ";
        cin>>input;
        towers[i] = input;
    }
    for(int i = number-1;i >=0;--i){
        int counter = 1;
        for(int j = i-1;j >= 0;--j){
            if(towers[i] >= towers[j]){
                ++counter;
            }
            else{
                break;
            }
        }
        signalPowers[i] = counter;
    }
    for(int i = 0;i < number;++i){
        cout<<signalPowers[i]<<" ";
    }

}

File: question3.py
#in this line we get our values
#you should get inputs with space between
towers = list(map(int,input("Please enter height of towers: ").split()))
#at first we make an empty list for saving singnal powers
signalPowers = []
while(len(towers) != 0):
    counter = 1
    #here we stimulated stack by poping the last elemnt
    Last_tower = towers.pop(len(towers)-1)
    for i in range(len(towers)-1,-1,-1):
        if towers[i] <= Last_tower:
            counter += 1
        else:
            break
    signalPowers.append(counter)
#we print our list from last element to the first
for i in range(len(signalPowers)-1,-1,-1):
    print(signalPowers[i],end = ' ')
        
        